editor/editor_properties.cpp

```diff
@@ -1959,16 +1959,17 @@ void EditorPropertyResource::_file_selected(const String &p_path) {
 
 	List<PropertyInfo> prop_list;
 	get_edited_object()->get_property_list(&prop_list);
-	String type;
+	String property_types;
 
 	for (List<PropertyInfo>::Element *E = prop_list.front(); E; E = E->next()) {
 		if (E->get().name == get_edited_property() && (E->get().hint & PROPERTY_HINT_RESOURCE_TYPE)) {
-			type = E->get().hint_string;
+			property_types = E->get().hint_string;
 		}
 	}
 
-	if (!type.empty() && !res->is_class(type))
-		EditorNode::get_singleton()->show_warning(vformat(TTR("The selected resource (%s) does not match the type expected for this property (%s)."), res->get_class(), type));
+	const Vector<String> split_property_types = property_types.split(",");
+	if (split_property_types.find(res->get_class()) == -1)
```

Comment:
A simple string compare won't work, as we need to match classes that inherit the type hint (e.g. `StreamTexture` is a valid `Texture` type). `Object::is_class` used previously does both check (== or inherits), so you should use that.

Reply:
My bad, thought it would be sufficient, should have checked Object::is_class before removing it for simple string comparison. I am on it.

Reply:
@akien-mga Corrected, now we check using is_class for every hinted type. I have also added check for an empty property_types string that I have removed in previous commit, in case the property doesn't have a resource type hint.